examples_utils/benchmarks/run_benchmarks.py

```diff
@@ -369,36 +360,44 @@ def run_benchmarks(args: argparse.ArgumentParser):
     with open(output_log_path, "w", buffering=1) as listener:
         logger.info(f"Logs at: {output_log_path}")
 
-        variant_dictionary = {}
-        for benchmark_name in sorted(spec.keys()):
+        # Only check explicitily listed benchmarks if provided
+        if args.benchmark is None:
+            args.benchmark = list(spec.keys())
+
+        variant_dictionary = OrderedDict()
+        for benchmark_name in args.benchmark:
+            # Check if this benchmark exists
+            if benchmark_name not in list(spec.keys()):
+                logger.error(f"Benchmark {benchmark_name} not found in " "provided spec files, exiting.")
+                sys.exit(1)
+
             # Do not treat the common options or similar specifications as
             # benchmarks
             if "options" in benchmark_name:
                 continue
 
+            # Skip convergence tests by default unless --include-convergence
+            # is provided, or they are explicitly named in --benchmarks
+            if ("_conv" in benchmark_name) and (not args.include_convergence):
+                continue
+
+            # Enforce DATASETS_DIR set only if this benchmark needs real data
+            if (not "gen" in benchmark_name) and (not "synth" in benchmark_name):
+                datasets_dir = os.getenv("DATASETS_DIR")
+                if datasets_dir is None:
+                    logger.error("Datasets directory has not been set.  If "
```

Comment:
You should raise an error instead of `sys.exit(1)`. I usually create the string and then log it with `logger.error` and then raise an Exception with the same string

Reply:
Thanks, will do from now on